src/main/kotlin/io/snyk/plugin/ui/jcef/OpenFileLoadHandlerGenerator.kt

```diff
@@ -35,32 +39,78 @@ class OpenFileLoadHandlerGenerator(snykCodeFile: SnykCodeFile) {
-                        browser.url, 0
-                    );
+                    val script = """
+                    (function() {
+                        if (window.openFileQuery) {
+                            return;
+                        }
+                        window.openFileQuery = function(value) { ${openFileQuery.inject("value")} };
+                        // Attach a single event listener to the document
+                        document.addEventListener('click', function(e) {
+                            // Find the nearest ancestor
+                            var target = e.target.closest('.data-flow-clickable-row');
+                            if (target) {
+                                e.preventDefault();
+                                window.openFileQuery(target.getAttribute("file-path") + ":" +
+                                                     target.getAttribute("start-line") + ":" +
+                                                     target.getAttribute("end-line") + ":" +
+                                                     target.getAttribute("start-character") + ":" +
+                                                     target.getAttribute("end-character"));
+                            }
+                        });
+                    })();
+                """
+                    browser.executeJavaScript(script, browser.url, 0)
+                    val colorTheme = if (EditorColorsManager.getInstance().isDarkEditor) "dark" else "light"
+                    val isDarculaTheme = isDarcula()
+                    val themeScript = """
+                        (function(){
+                        if (window.themeApplied) {
+                            return;
+                        }
+                        window.themeApplied = true;
+                        const style = getComputedStyle(document.documentElement);
+                        const properties = [
+                          '--text-color',
+                          '--background-color',
+                          '--link-color',
+                          '--info-text-color',
+                          '--disabled-text-color',
+                          '--selected-text-color',
+                          '--error-text-color',
+                          '--data-flow-file-color',
+                          '--data-flow-body-color',
+                          '--example-line-added-color',
+                          '--example-line-removed-color',
+                          '--tabs-bottom-color',
+                          '--tab-item-color',
+                          '--tab-item-hover-color',
+                          '--tab-item-icon-color',
+                          '--scrollbar-thumb-color',
+                          '--scrollbar-color',
+                        ]
+                        properties.forEach(p => document.documentElement.style.setProperty(p, style.getPropertyValue(p + "-" + "${colorTheme}")))
```

Comment:
@bastiandoetsch about your comment:

>Regarding the if (isDarktheme)... I'd just define a variable theme that you can add, so you don't clutter the styles with inline if clauses.

Have a look at the following refactor, please and let me know what you thing :)

Reply:
Yeah, much better. Still hoping for a more generic solution though. I feel uneasy having specific themes in our extension design - would expect that IntelliJ provides all styles from the theme, and we can just reference them (color, font, etc).

Reply:
Sorry @bastiandoetsch , maybe I'm not understanding what you mean. The development so far when moving the HTML and CSS to LSP is like "server-side rendering":

- The initial template and styles are prepared before they reach the client (the IDE in this case)
- Once delivered, any dynamic adjustments based on the IDE settings or themes are handled client-side

Then, once the page is loaded in the IDE, IntelliJ in this case, we still need to apply the [JetBrains typography](https://jetbrains.design/intellij/principles/typography/#colors) from the Kotlin code with `EditorColorsManager.getInstance().globalScheme`. We dynamically update the CSS variables in the HTML to match these styles.

Do you have any other suggestion? Maybe I missed exploring other alternatives 🤔